pkg/commonvpc/vpc.go

```diff
@@ -179,15 +190,74 @@ func (s *Subnet) AddressRange() *ipblock.IPBlock {
 	return s.IPblock
 }
 func (s *Subnet) IsPrivate() bool {
-	return s.isPrivate
+	// dontcare means that the provider does not allow to set the subnet to be private, IsPrivate() will return false
+	return s.subnetExpose == privateExpose
 }
+
+// SetIsPrivate() is called only for platforms that support private/public subnets.
+// in other cases, the value of s.subnetExpose remain dontcare
 func (s *Subnet) SetIsPrivate(isPrivate bool) {
-	s.isPrivate = isPrivate
+	s.subnetExpose = publicExpose
+	if isPrivate {
+		s.subnetExpose = privateExpose
+	}
 }
 func (s *Subnet) SynthesisKind() spec.ResourceType {
 	return spec.ResourceTypeSubnet
 }
 
+// ////////////////////////////////////////////////////////////////////////////////
+// privateSubnetRule is the implementation of PrivateSubnetRule
+// it holds the information on the influence of the subnet on the connectivity.
+// the rule is created only in case that the subnet configuration has influence on the connectivity between src and dst
+// i.e. its relevant only for providers that allow private subnets (aws), and one of the nodes is external
+type privateSubnetRule struct {
+	subnet    vpcmodel.Subnet
+	src, dst  vpcmodel.Node
+	isIngress bool
+}
+
+func newPrivateSubnetRule(subnet vpcmodel.Subnet, src, dst vpcmodel.Node, isIngress bool) vpcmodel.PrivateSubnetRule {
+	return &privateSubnetRule{subnet, src, dst, isIngress}
+}
+
+// Note that this func is called only when relevant (platform supporting private subnet and connection to/from internet)
+func (psr *privateSubnetRule) Deny(isIngress bool) bool {
+	return isIngress == psr.isIngress && psr.subnet.IsPrivate()
+}
+
+func (psr *privateSubnetRule) String() string {
+	switch {
+	case psr.Deny(false):
+		return fmt.Sprintf("%s will not accept connection from %s, since subnet %s is private\n",
+			psr.dst.Name(), psr.src.Name(), psr.subnet.Name())
+	case psr.Deny(true):
+		return fmt.Sprintf("%s will not connect to %s, since subnet %s is private\n",
+			psr.src.Name(), psr.dst.Name(), psr.subnet.Name())
+	case !psr.isIngress:
+		return fmt.Sprintf("%s can accept connection from %s, since subnet %s is public\n",
+			psr.dst.Name(), psr.src.Name(), psr.subnet.Name())
+	case psr.isIngress:
+		return fmt.Sprintf("%s can connect to %s, since subnet %s is public\n",
+			psr.src.Name(), psr.dst.Name(), psr.subnet.Name())
```

Comment:
```suggestion
	default:
		return fmt.Sprintf("traffic from %s to %s is allowed, since subnet %s is public\n",
			psr.src.Name(), psr.dst.Name(), psr.subnet.Name())
```

Reply:
same here

Reply:
same here - lets insert Ziv's suggestions @haim-kermany